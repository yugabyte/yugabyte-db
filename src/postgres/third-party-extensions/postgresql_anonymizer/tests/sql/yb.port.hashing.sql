
/* YB: workaround for lack of transactional DDL
BEGIN;
*/ -- YB

CREATE EXTENSION IF NOT EXISTS anon CASCADE;

BEGIN; -- YB: Workaround for read time error, check #25665
SET yb_non_ddl_txn_for_sys_tables_allowed = true; -- YB: next statement updates pg_seclabel and is not a DDL
SELECT anon.init();
COMMIT; -- YB: Workaround for read time error, check #25665

--
-- Using the automatic salt and default algorithm
--
SELECT anon.hash(NULL) IS NULL;

SELECT anon.hash('x')
     = anon.digest('x',
                    pg_catalog.current_setting('anon.salt'),
                    pg_catalog.current_setting('anon.algorithm')
                 );

--
-- With a random salt
--
SELECT anon.random_hash('abcd') != anon.random_hash('abcd');

-- Restore a predifened salt and change the algo
SET anon.salt TO '4a6821d6z4e33108gg316093e6182b803d0361';
SET anon.algorithm TO 'md5';
SELECT anon.hash('x');
SET anon.algorithm TO 'sha512';
SELECT anon.hash('x');

-- digest
SELECT anon.digest(NULL,'b','sha512') IS NULL;
SELECT anon.digest('a',NULL,'sha512') IS NULL;
SELECT anon.digest('a','b',NULL) IS NULL;

SELECT anon.digest('a','b','md5') = '187ef4436122d1cc2f40dc2b92f0eba0';
SELECT anon.digest('a','b','sha1') IS NULL;
SELECT anon.digest('a','b','sha224') = 'db3cda86d4429a1d39c148989566b38f7bda0156296bd364ba2f878b';
SELECT anon.digest('a','b','sha256') = 'fb8e20fc2e4c3f248c60c39bd652f3c1347298bb977b8b4d5903b85055620603';
SELECT anon.digest('a','b','sha384') = 'c7be03ba5bcaa384727076db0018e99248e1a6e8bd1b9ef58a9ec9dd4eeebb3f48b836201221175befa74ddc3d35afdd';
SELECT anon.digest('a','b','sha512') = '2d408a0717ec188158278a796c689044361dc6fdde28d6f04973b80896e1823975cdbf12eb63f9e0591328ee235d80e9b5bf1aa6a44f4617ff3caf6400eb172d';

/* YB: workaround for lack of transactional DDL
ROLLBACK;
*/ -- YB

DROP EXTENSION anon CASCADE; -- YB: workaround for lack of transactional DDL
