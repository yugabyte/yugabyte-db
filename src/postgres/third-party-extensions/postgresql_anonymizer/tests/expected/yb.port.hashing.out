/* YB: workaround for lack of transactional DDL
BEGIN;
*/ -- YB
CREATE EXTENSION IF NOT EXISTS anon CASCADE;
BEGIN; -- YB: Workaround for read time error, check #25665
SET yb_non_ddl_txn_for_sys_tables_allowed = true; -- YB: next statement updates pg_seclabel and is not a DDL
SELECT anon.init();
 init 
------
 t
(1 row)

COMMIT; -- YB: Workaround for read time error, check #25665
--
-- Using the automatic salt and default algorithm
--
SELECT anon.hash(NULL) IS NULL;
 ?column? 
----------
 t
(1 row)

SELECT anon.hash('x')
     = anon.digest('x',
                    pg_catalog.current_setting('anon.salt'),
                    pg_catalog.current_setting('anon.algorithm')
                 );
 ?column? 
----------
 t
(1 row)

--
-- With a random salt
--
SELECT anon.random_hash('abcd') != anon.random_hash('abcd');
 ?column? 
----------
 t
(1 row)

-- Restore a predifened salt and change the algo
SET anon.salt TO '4a6821d6z4e33108gg316093e6182b803d0361';
SET anon.algorithm TO 'md5';
SELECT anon.hash('x');
               hash               
----------------------------------
 9072b9a6712c6d3924c79488d6ee1c2f
(1 row)

SET anon.algorithm TO 'sha512';
SELECT anon.hash('x');
                                                               hash                                                               
----------------------------------------------------------------------------------------------------------------------------------
 ef50c6f79ba70b3eb96cded76956ef59f9aa2ff0365b7ccec6d128d21bc52be21236b72a90a77c9147b542eaacbe90a7134bc4477d5774ba7ac1443764147b15
(1 row)

-- digest
SELECT anon.digest(NULL,'b','sha512') IS NULL;
 ?column? 
----------
 t
(1 row)

SELECT anon.digest('a',NULL,'sha512') IS NULL;
 ?column? 
----------
 t
(1 row)

SELECT anon.digest('a','b',NULL) IS NULL;
 ?column? 
----------
 t
(1 row)

SELECT anon.digest('a','b','md5') = '187ef4436122d1cc2f40dc2b92f0eba0';
 ?column? 
----------
 t
(1 row)

SELECT anon.digest('a','b','sha1') IS NULL;
 ?column? 
----------
 t
(1 row)

SELECT anon.digest('a','b','sha224') = 'db3cda86d4429a1d39c148989566b38f7bda0156296bd364ba2f878b';
 ?column? 
----------
 t
(1 row)

SELECT anon.digest('a','b','sha256') = 'fb8e20fc2e4c3f248c60c39bd652f3c1347298bb977b8b4d5903b85055620603';
 ?column? 
----------
 t
(1 row)

SELECT anon.digest('a','b','sha384') = 'c7be03ba5bcaa384727076db0018e99248e1a6e8bd1b9ef58a9ec9dd4eeebb3f48b836201221175befa74ddc3d35afdd';
 ?column? 
----------
 t
(1 row)

SELECT anon.digest('a','b','sha512') = '2d408a0717ec188158278a796c689044361dc6fdde28d6f04973b80896e1823975cdbf12eb63f9e0591328ee235d80e9b5bf1aa6a44f4617ff3caf6400eb172d';
 ?column? 
----------
 t
(1 row)

/* YB: workaround for lack of transactional DDL
ROLLBACK;
*/ -- YB
DROP EXTENSION anon CASCADE; -- YB: workaround for lack of transactional DDL
